pkg/resource/v1/chartconfig/key/key.go

```diff
@@ -0,0 +1,31 @@
+package key
+
+import (
+	"strings"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/microerror"
+)
+
+// ClusterID returns cluster ID for given guest cluster config.
+func ClusterID(clusterGuestConfig v1alpha1.ClusterGuestConfig) string {
+	return clusterGuestConfig.ID
+}
```

Comment:
These key helpers are also in the other resources in `pkg/resource/v1`. Do we need `pkg/resource/v1/key` for these?

Reply:
@tuommaki created this initially. I do not follow this approach here.

Reply:
I think now that it seems that all shared resource implementations can settle with `ClusterGuestConfig`, maybe it makes sense to combine current `key` packages from resources to `pkg/resource/v1/key` as Ross suggests. Only remaining thing I've subconsciously tried to solve is the problem of getting namespace of provider specific CR to shared resources, but maybe this requires another extraction function. This namespace is needed so that created resources can be created to same namespace with original `cluster-operator` CR.

Reply:
As noted in other comment, this would probably be reasonable to move `pkg/resource/v1/key` as suggested.

Reply:
I created a PR for this: https://github.com/giantswarm/cluster-operator/pull/44

Reply:
I think moving key up in the package structure for reuse is a good idea. I'll integrate with #44 once it's merged.